drivers/cc110x/cc110x-spi.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_cc110x
+ * @{
+ *
+ * @file
+ * @brief       TI Chipcon CC110x spi driver
+ *
+ * @author      Thomas Hillebrandt <hillebra@inf.fu-berlin.de>
+ * @author      Heiko Will <hwill@inf.fu-berlin.de>
+ * @author      Fabian Nack <nack@inf.fu-berlin.de>
+ * @author      Joakim Gebart <joakim.gebart@eistec.se>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "cc110x.h"
+#include "cc110x/cc110x-spi.h"
+#include "cc110x/cc110x-internal.h"
+#include "cc110x/cc110x-defines.h"
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#include "hwtimer.h"
+#include "irq.h"
+
+/**********************************************************************
+ *                      CC110x spi access
+ **********************************************************************/
+
+void cc110x_cs(cc110x_t *dev)
+{
+    volatile int retry_count = 0;
+//    /* Switch MISO/GDO1 to GPIO input mode */
+    gpio_init(dev->params.GDO1, GPIO_DIR_IN, GPIO_NOPULL);
+    /* CS to low */
+    gpio_clear(dev->params.CS);
+    /* Wait for SO to go low (voltage regulator
+     * has stabilized and the crystal is running) */
+    while (gpio_read(dev->params.GDO1)) {
+        /* Wait ~500us and try again */
+        hwtimer_wait(HWTIMER_TICKS(CS_SO_WAIT_TIME));
+
+        if (gpio_read(dev->params.GDO1)) {
+            retry_count++;
+
+            if (retry_count > CC110X_GDO1_LOW_RETRY) {
+                puts("[CC110X spi] fatal error\n");
+                break;
+            }
+
+            gpio_set(dev->params.CS);
+            gpio_clear(dev->params.CS);
+        }
+    }
+    /* Switch MISO/GDO1 to spi mode */
+    //spi_conf_pins(dev->params.spi);
+}
+
+void cc110x_writeburst_reg(cc110x_t *dev, uint8_t addr, const char *src, uint8_t count)
+{
+    unsigned int cpsr;
+    spi_acquire(dev->params.spi);
+    cpsr = disableIRQ();
```

Comment:
why are you disabling interrupts during transfer? The transfer should be thread safe, so no need?!